### ISSUE TITLE
imgur images failing to load on google spreadsheet timeline

### ISSUE THREAD

**Issue by answerquest** [NUKnightLab/TimelineJS3#545]
Here is a [link to the spreadsheet](https://docs.google.com/spreadsheets/d/1wh3OZzNMoGU3rrQqPJwQb39bEBHYkzLk6zWr7AlHffc/edit?usp=sharing).

And here is a [link to the page](https://answerquest.github.io/timeline1.html) where the timeline is embedded.

Console error logged: 
`Failed to load resource: the server responded with a status of 403 (Forbidden)`

..twice for two imgur direct links: <https://i.imgur.com/81vHNWGl.png> and <https://i.imgur.com/sxe1CeMm.png> .

If the said images have already been loaded earlier on the browser by directly opening their URL or seeing them on imgur, then they load in the timeline. But if the browser hasn't loaded them earlier, then they seem to error out. The documentation states that direct links to images hosted on imgur can be used. Please see if this error happens on anyone else's end too. I'm using chromium / firefox browser on a Lubuntu 16.04 system.

**Comment by scott2b** [comment]
This is an increasingly common problem with imgur. There is nothing really we can do to prevent them from serving 403s on content hosted by them. Timeline's link resolution is not the problem here. It is restrictions from Imgur. I have added a note to the media types documentation on the Timeline site

**Comment by answerquest** [comment]
Ok.. thanks for your time. Really appreciate the whole thing, you guys have made something beautiful.

**Comment by lderas** [comment]
2020 and I'm still getting 403 on my regular imgur links.

**Comment by scott2b** [comment]
Imgur seems to be real hit or miss on cross origin requests. I tend to recommend that people not use it for our tools for this reason

**Comment by JoeGermuska** [comment]
**tl;dr: don't use Imgur "direct" image links in Timeline; use Imgur page ("gallery") URLs instead.**

This problem occurs because [Imgur does not support general "hotlinking" of images](https://help.imgur.com/hc/en-us/articles/201569776?input_string=hotlinking+to+imgur+images):
> We allow hotlinking on forums, but hotlinked images cannot be used as content for a website, including blog posts, avatars, site elements, and advertising. You can read our terms of service for more info. If you believe your site has been mistakenly blocked, please contact us for assistance.

The 403 errors are served when Imgur enforces this policy. If the Imgur images occasionally appear (as described in the initial description of this issue), it must be because Imgur systems don't always serve responses through a codepath that enforces the policy.  

The correct way for people to embed Imgur images is to use the Imgur page URL (eg https://imgur.com/gallery/NEzPd3N) not the direct image URL (https://i.imgur.com/S1ipxA4.jpg). This ensures that Timeline's "media type" handler for Imgur operates instead of the "media type" handler for direct image links.  

A side effect of this is that the image will be embedded in Imgur "chrome" like this:
<img width="551" alt="image" src="https://user-images.githubusercontent.com/46313/89195497-738d9980-d56e-11ea-9cca-0a9ae9aa812f.png">

This is part of Imgur's terms of service, and not something we have any choice in.  Another side effect of this is that Imgur images are not suitable for use as slide backgrounds or media thumbnails, because they can't truly be served as images.

If there were a reliable relationship between the direct image URLs and the gallery pages, we might be able to intercept Imgur direct image links and handle them, but as can be seen above, there's no clear mapping between the direct image link and the page ("gallery") URL, and Imgur's oEmbed handler which provides the embed we use doesn't work with direct image links, nor does it work if the identifier part of the direct image link is extracted and used to construct a "gallery" URL.
